src/main/java/com/sda/zdjavapol107/statki/ship/ShipService.java

```diff
@@ -0,0 +1,261 @@
+package com.sda.zdjavapol107.statki.ship;
+
+import com.sda.zdjavapol107.statki.MessagesPrinter;
+import com.sda.zdjavapol107.statki.model.PointShipsGame;
+
+import java.util.ArrayList;
+import java.util.InputMismatchException;
+import java.util.List;
+import java.util.Scanner;
+
+public class ShipService {
+
+    private ShipDirection shipDirectionEnum;
+    private boolean isShipPositionOk = false;
+    private int choiceShipSize = 1;
+    protected int fourSizeShip = 1;
+    protected int threeSizeShip = 2;
+    protected int twoSizeShip = 3;
+    protected int oneSizeShip = 4;
+    public int shipDirection;
+    private boolean endOfPlacesTheShips = false;
+    private PointShipsGame firstPointShip;
+    private int userPointY;
+    private int userPointX;
+    private boolean endLoopCreatingShip;
+    public List<PointShipsGame> busyPointsOnBoard = new ArrayList<>();
+    private MessagesPrinter messagesPrinter = new MessagesPrinter();
+    private Scanner scanner = new Scanner(System.in);
+
+    // Pokauzje infomracje ile statkow zostalo do postawienia przez uzytkownika
+    public void creatingShip() {
+        boolean haveShip = true;
+        do {
+            do {
+                messagesPrinter.lastShipsToPlace(fourSizeShip, threeSizeShip, twoSizeShip, oneSizeShip);
+                try { // Czy jest jeszcze dostepny wybrany statek przez uzytkownika do postawienia na planszy
+                    choiceShipSize = scanner.nextInt();
+                    if (choiceShipSize > 0 && choiceShipSize < 5 && checkAvilabilityShips()) {
+                        placesTheSingleShip(choiceShipSize);
+                        if (fourSizeShip == 0 && threeSizeShip == 0 && twoSizeShip == 0 && oneSizeShip == 0) {
+                            endOfPlacesTheShips = true;
+                        }
+                    }
+                } catch (InputMismatchException e) {
+                    messagesPrinter.errorInputValue();
+                }
+            } while (endOfPlacesTheShips == false);
+        } while (endOfPlacesTheShips == false);
+    }
+
+    private boolean checkAvilabilityShips() { // To samo co u gory
+        boolean haveShip = true;
+        if (choiceShipSize == 1) {
```

Comment:
jak widzicie ten sam if 4 razy tylko z roznymi parametrami to nie mysleliscie moze zeby zrobic funkcji dla ktorej przekazecie te parametry ktore sie zmieniaja :D?